presto-main/src/main/java/io/prestosql/operator/window/FirstValueFunction.java

```diff
@@ -41,24 +51,26 @@ public void processRow(BlockBuilder output, int frameStart, int frameEnd, int cu
             output.appendNull();
             return;
         }
-
-        int valuePosition = frameStart;
+        int returnInd;
 
         if (ignoreNulls) {
-            while (valuePosition >= 0 && valuePosition <= frameEnd) {
-                if (!windowIndex.isNull(argumentChannel, valuePosition)) {
-                    break;
-                }
-
-                valuePosition++;
-            }
-
-            if (valuePosition > frameEnd) {
-                output.appendNull();
-                return;
-            }
+            returnInd = searchWindowAndUpdateIndices(frameStart, frameEnd);
+        }
+        else {
+            returnInd = frameStart;
         }
+        windowIndex.appendTo(argumentChannel, returnInd, output);
+    }
```

Comment:
I would extract this if block as method so you don't need to have uninitialized variable

Reply:
Refactored to avoid uninitialized returnInd (didn't introduce a method though).